### PR TITLE
Add processor for Tinkerbell Template Config

### PR DESCRIPTION
**Pull request by mitalipaygude**
*Issue #, if available:*
https://github.com/aws/eks-anywhere/issues/7816

*Description of changes:*
Add processor for Tinkerbell Template Config

*Testing (if applicable):*
- unit test
- controller image testing

*Documentation added/planned (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/cluster/tinkerbell.go [hunk 1]
```diff
@@ -125,6 +125,16 @@ func getTinkerbellMachineConfigs(ctx context.Context, client Client, c *Config)
 		}
 
 		c.TinkerbellMachineConfigs[machineConfig.Name] = machineConfig
+
+		if !machineConfig.Spec.TemplateRef.IsEmpty() {
+			c.TinkerbellTemplateConfigs = map[string]*anywherev1.TinkerbellTemplateConfig{}
```
Comment (pokearu) [inline pkg/cluster/tinkerbell.go:130]: wont this get reset for each template config? So only the last TemplateRef will get set. We can have different templateRefs for each machineconfig right?

Comment (mitalipaygude) [reply pkg/cluster/tinkerbell.go:130]: oh yeah, good catch. Fixing it!

**Review by pokearu — APPROVED** [review]
/lgtm
/meow

**Pull request merged**